docroot/modules/custom/uiowa_core/uiowa_core.module

```diff
@@ -134,6 +135,155 @@ function template_preprocess_uiowa_core_headline(&$variables) {
+  if (!$is_figure && isset($variables['elements']['#embed']) && $variables['elements']['#embed']) {
+
+    // If we have a view mode set...
+    if (isset($variables['view_mode'])) {
+
+      // If we have relevant view mode classes...
+      if (!is_null($add_class = uiowa_core_get_inline_size_class_from_view_mode($variables['view_mode']))) {
+
+        // Attach the inline sizing library and assign the classes.
+        $variables['#attached']['library'][] = 'uids_base/inline';
+        $variables['attributes']['class'][] = $add_class;
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess_filter_caption().
+ */
+function uiowa_core_preprocess_filter_caption(&$variables) {
+
+  // Get an associative array of attributes from the `<drupal-media>` node.
+  $attributes = uiowa_core_drupal_media_attributes($variables['node']);
+
+  $inline_class = 'inline--size-default';
+
+  // If we have a view mode set...
+  if (isset($attributes['data-view-mode'])) {
+    // Process the view mode classes.
+    // Check if the inline class needs to be changed.
+    if (!is_null($override_inline_class = uiowa_core_get_inline_size_class_from_view_mode($attributes['data-view-mode']))) {
+
+      // Replace the $inline_class with the relevant
+      // modified inline view mode class.
+      $inline_class = $override_inline_class;
+    }
+  }
+  // Get the current classes of the '<drupal-media>' element.
+  $classes_array = explode(' ', $variables['classes']);
+
+  // Add the view mode to the classes.
+  $classes_array[] = $inline_class;
+
+  // Re-concatenate the $classes_array back in to a string,
+  // which is what the '<drupal-media>' element wants.
+  $classes = implode(' ', $classes_array);
+
+  // Attach the inline sizing library and assign the classes.
+  $variables['#attached']['library'][] = 'uids_base/inline';
+  $variables['classes'] = $classes;
+
+  // Then, add a custom attribute to the '<drupal-media>'
+  // markup to show that it is a figure.
+  $variables['node'] = FilteredMarkup::create(preg_replace(
+    '/<drupal-media /i',
+    '<drupal-media data-figure="true" ',
+    $variables['node']));
+}
+
+/**
+ * Provides mapping for a $view_mode to an inline view mode.
+ *
+ * @param string $view_mode
+ *   A string that is going to be checked
+ *   as a view mode, often looks something
+ *   like 'medium__no_crop'.
+ *
+ * @return string|null
+ *   If the $view_mode starts with any one of the $accepted_view_modes:
+ *    Returns an array containing the string reference to
+ *    the inline library and the inline sizing class.
+ *   Otherwise, returns an empty array.
+ */
+function uiowa_core_get_inline_size_class_from_view_mode(string $view_mode): ?string {
+  $accepted_view_modes = ['default', 'large', 'medium', 'small'];
+
+  foreach ($accepted_view_modes as $value) {
+    if (str_starts_with($view_mode, $value)) {
+
+      // Return the class that needs to be added.
+      return "inline--size-$value";
+    }
+  }
+
+  return NULL;
+}
```

Comment:
```suggestion
function uiowa_core_get_inline_size_class_from_view_mode(string $view_mode): ?string {
  // The view mode sizes which have a matching class. Defining them here
  // will define what we use in the regex matching pattern.
  $accepted_view_modes = ['default', 'large', 'medium', 'small'];

  // Look for any of our accepted view modes by checking the
  // start of the view mode string.
  // Regex101 explanation: https://regex101.com/r/CHe1tD/1
  // The matched $value will be an array, but we only need the first entry,
  // as it contains the full match we're interested in.
  if (preg_match('%^(' . implode('|', $accepted_view_modes) . ')%i', $view_mode, $value)) {
    return [
      'library' => 'uids_base/inline',
      'class' => 'inline--size-' . $value[0]
    ];
  }

  return NULL;
}
```

Reply:
@cory-skeers This suggestion is returning an `?array` instead of `?string`.

Reply:
Updated to only return the string.